drivers/png/image_loader_png.cpp

```diff
@@ -239,7 +243,8 @@ Error ImageLoaderPNG::_load_image(void *rf_up, png_rw_ptr p_func, Image *p_image
 }
 
 Error ImageLoaderPNG::load_image(Image *p_image, FileAccess *f) {
-
+	
+	file = f;
```

Comment:
If something must be stored, it's simpler to just store the file path and not the whole FileAccess.
Also, please squash commits together.